pkg/controller/master-controller-manager/user-synchronizer/controller.go

```diff
@@ -84,39 +82,14 @@ func Add(
 	}
 
 	if err := c.Watch(
-		&source.Kind{Type: &kubermaticv1.User{}}, &handler.EnqueueRequestForObject{}, serviceAccountPredicate, withEventFilter(),
+		&source.Kind{Type: &kubermaticv1.User{}}, &handler.EnqueueRequestForObject{}, serviceAccountPredicate,
 	); err != nil {
 		return fmt.Errorf("failed to create watch for user objects in master cluster: %w", err)
 	}
 
 	return nil
 }
 
-func withEventFilter() predicate.Predicate {
-	return predicate.Funcs{
-		CreateFunc: func(e event.CreateEvent) bool {
-			return true
-		},
-		UpdateFunc: func(e event.UpdateEvent) bool {
```

Comment:
what about changing `UpdateFunc` to:
```
return e.ObjectOld.GetGeneration() != e.ObjectNew.GetGeneration()
```
this way we won't get events about `status` subresource update. 

I could imagine that we just compare `spec` without `metadata` and it's causing an issue, but still have a problem to understand how it happens like i described [here](https://github.com/kubermatic/kubermatic/pull/8288/files#r756157812)

Reply:
ok, now I understand the update with `deletionTimestamp` as @lsviben pointed

Reply:
so @imharshita we should change the update func to 
```
return e.ObjectOld.GetGeneration() != e.ObjectNew.GetGeneration()
```
to filter out updates to status fields